clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -6845,3 +6847,57 @@ ExprResult SemaSYCL::ActOnUniqueStableNameExpr(SourceLocation OpLoc,
 
   return BuildUniqueStableNameExpr(OpLoc, LParen, RParen, TSI);
 }
+
+void SemaSYCL::performSYCLDelayedAttributesAnalaysis(const FunctionDecl *FD) {
+  if (UserProvidedSYCLKernelFunctions.contains(FD))
+    return;
+
+  auto isDependent =
```

Comment:
Why the dependency checks?

Reply:
Issuing that diagnostic happens twice without checking dependency in templated cases.

Reply:
What is FD in each of these cases? I assume one is the general template and one is the instantiation. Rather than dependency checks on the attributes, I think you can check FD.

Reply:
I will check it